common/src/service/IotHubConnectionString.cs

```diff
@@ -11,39 +11,29 @@
 
 namespace Microsoft.Azure.Devices
 {
-    internal sealed class IotHubConnectionString : IAuthorizationHeaderProvider, ICbsTokenProvider
+    /// <summary>
+    /// The properties required for authentication to IoT hub using a connection string.
+    /// </summary>
+    internal sealed class IotHubConnectionString : IotHubCredential
     {
-        private static readonly TimeSpan s_defaultTokenTimeToLive = TimeSpan.FromHours(1);
-        private const char UserSeparator = '@';
+        private static readonly TimeSpan _tokenTimeToLive = TimeSpan.FromHours(1);
 
-        public IotHubConnectionString(IotHubConnectionStringBuilder builder)
+        public IotHubConnectionString(IotHubConnectionStringBuilder builder) : base(builder.HostName)
         {
             if (builder == null)
```

Comment:
we're checking for builder null after the 'base(builder.HostName)' call. Maybe move the base(builder.HostName) call after the null check.

Reply:
Makes sense. Fixing it in next PR itself as it almost ready to go. I don't want to run the gates again :D